internal/service/appmesh/route.go

```diff
@@ -375,13 +387,35 @@ func ResourceRoute() *schema.Resource {
 																Required:     true,
 																ValidateFunc: validation.IntBetween(0, 100),
 															},
+
+															"port": {
+																Type:         schema.TypeInt,
+																Optional:     true,
+																ValidateFunc: validation.IntBetween(1, 65535),
```

Comment:
This should use already existing `validation.IsPortNumber()`. E.g. https://github.com/hashicorp/terraform-provider-aws/pull/27368/files#diff-344a72a045e6eba371d6884f2e94133929be147a955a19bb342c4018b756660fR498